pandas/core/arrays/datetimes.py

```diff
@@ -210,36 +234,98 @@ class DatetimeArrayMixin(dtl.DatetimeLikeArrayMixin,
     # Constructors
 
     _attributes = ["freq", "tz"]
```

Comment:
if the signature for _simple_new is changed to take dtype instead of tz, then i think this needs to be changed too.